demisto_sdk/commands/download/downloader.py

```diff
@@ -376,32 +390,41 @@ def fetch_custom_content(self) -> bool:
             tar = tarfile.open(fileobj=io_bytes, mode="r")
 
             scripts_id_name: dict = {}
+            strings_to_write: List[Tuple[str, str]] = []
             for member in tar.getmembers():
                 file_name: str = self.update_file_prefix(member.name.strip("/"))
                 file_path: str = os.path.join(self.custom_content_temp_dir, file_name)
 
                 extracted_file = tar.extractfile(member)
-                # File might empty
                 if extracted_file:
                     string_to_write = extracted_file.read().decode("utf-8")
-                    string_to_write, scripts_id_name = self.handle_file(
-                        string_to_write, member.name, scripts_id_name
-                    )
-
-                    try:
-                        with open(file_path, "w") as file:
-                            file.write(string_to_write)
-
-                    except Exception as e:
-                        print(f"encountered exception {type(e)}: {e}")
-                        print("trying to write with encoding=utf8")
-                        with open(file_path, "w", encoding="utf8") as file:
-                            file.write(string_to_write)
+                    if (
+                        "automation-" in member.name.lower()
+                        or "integration-" in member.name.lower()
+                    ):
+                        scripts_id_name = self.map_script(
+                            string_to_write, scripts_id_name
+                        )
+                    strings_to_write.append((string_to_write, member.name))
                 else:
                     raise FileNotFoundError(
                         f"Could not extract files from tar file: {file_path}"
                     )
 
+            for tup in strings_to_write:
```

Comment:
`tup` is ambiguous, better do `for file_name, file_type in strings_to_write` (if that's how the tuples are built)